### PR TITLE
asBuffer fix

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/encoding-down/issues/18

### CODE REVIEW

**Comment by vweevers** [comment]
Do we have tests for the `keyAsBuffer` and `valueAsBuffer` options on iterators?

**Comment by ralphtheninja** [comment]
> Do we have tests for the keyAsBuffer and valueAsBuffer options on iterators?

Added some tests.

File: test/index.js [hunk 1]
```diff
@@ -108,3 +109,79 @@ test('binary encoding, using batch', function (t) {
...
+  t.plan(2)
+
+  var down = {
+    iterator: function (options) {
+      t.is(options.keyAsBuffer, false, '.keyAsBuffer should be false')
+      t.is(options.valueAsBuffer, false, '.valueAsBuffer should be false')
+    }
+  }
+
+  encdown(down).iterator({ keyAsBuffer: false, valueAsBuffer: false })
```
Comment (vweevers) [inline test/index.js]: Can you write this in the same way as the `get()` test? Becauxe the `keyAsBuffer` and `valueAsBuffer` options should also be based on `encoding.buffer`. 

We used to have this code in `levelup`:

```js
options = this._codec.encodeLtgt(options)		
options.keyAsBuffer = this._codec.keyAsBuffer(options)		
options.valueAsBuffer = this._codec.valueAsBuffer(options)
```

Comment (ralphtheninja) [reply test/index.js]: Sure!

Comment (ralphtheninja) [reply test/index.js]: If you don't mind, I'll add an _additional_ test and keep this one as it is.

File: test/index.js [hunk 2]
```diff
@@ -159,3 +159,29 @@ test('custom value encoding that retrieves a buffer from underlying store', func
...
   db.get('key', noop)
 })
+
+test('iterator options .keyAsBuffer and .valueAsBuffer defaults', function (t) {
+  t.plan(2)
+
+  var down = {
+    iterator: function (options) {
+      t.is(options.keyAsBuffer, true, '.keyAsBuffer defaults to true')
+      t.is(options.valueAsBuffer, true, '.valueAsBuffer defaults to true')
```
Comment (ralphtheninja) [inline test/index.js]: I just noticed this is wrong since `utf8` is default encoding and should produce `buffer: false`

**Comment by ralphtheninja** [comment]
@vweevers This should be better I think. Will squash afterwards.

File: test/index.js (hunk 1 above)
Comment (vweevers) [reply test/index.js]: Got me thinking. Big picture, if you use `encoding-down`, then `*asBuffer` is an internal property determined by encoding. But if you don't, then `asBuffer` is a public option, like in [this `levelup` test](https://github.com/Level/levelup/blob/a4cd129b3eb3ceb15d798137fefd531651ffa070/test/no-encoding-test.js#L33). Perhaps that needs clarification in docs (eventually, as most people *will* use `encoding-down`), but where?

Comment (ralphtheninja) [reply test/index.js]: @vweevers Good point. The idea struck my mind as well. Make an issue for it for now so we can move out the discussion from this PR?

**Pull request merged**